### PR TITLE
fix: forwardRef render functions not using ref

### LINKED ISSUE #4167: [BUG] - Autocomplete forwardRef Error in 2.5.0 Beta
### NextUI Version

2.5.0-beta.24

### Describe the bug

With React 19 and Nextjs 15.0.3 and the latest beta for 2.5.0-beta.24 I am getting error for the Autocomplete component.

This seems to come from the Listbox inside Autocomplete.

Full Error log:

Error: forwardRef render functions accept exactly two parameters: props and ref. Did you forget to use the ref parameter?
    at createUnhandledError (http://localhost:3000/_next/static/chunks/08b5e_next_dist_client_9bb038._.js:744:19)
    at handleClientError (http://localhost:3000/_next/static/chunks/08b5e_next_dist_client_9bb038._.js:905:98)
    at error (http://localhost:3000/_next/static/chunks/08b5e_next_dist_client_9bb038._.js:1036:56)
    at console.error (https://eu-assets.i.posthog.com/static/recorder.js?v=1.193.1:1:103627)
    at exports.forwardRef (http://localhost:3000/_next/static/chunks/08b5e_next_dist_compiled_1359e6._.js:1568:418)
    at forwardRef (http://localhost:3000/_next/static/chunks/08b5e_b5d0a5._.js:6120:204)
    at [project]/node_modules/@nextui-org/listbox/dist/chunk-W7VMXJNC.mjs [app-client] (ecmascript) (http://localhost:3000/_next/static/chunks/08b5e_@nextui-org_0a72b7._.js:7883:231)
    at http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:693:27
    at runModuleExecutionHooks (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:738:9)
    at instantiateModule (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:691:9)
    at getOrInstantiateModuleFromParent (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:624:12)
    at esmImport (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:142:20)
    at [project]/node_modules/@nextui-org/listbox/dist/chunk-Y2D6DCKD.mjs [app-client] (ecmascript) (http://localhost:3000/_next/static/chunks/08b5e_@nextui-org_0a72b7._.js:7901:179)
    at http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:693:27
    at runModuleExecutionHooks (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:738:9)
    at instantiateModule (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:691:9)
    at getOrInstantiateModuleFromParent (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:624:12)
    at esmImport (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:142:20)
    at [project]/node_modules/@nextui-org/listbox/dist/chunk-Y2D6DCKD.mjs [app-client] (ecmascript) <export listbox_default as Listbox> (http://localhost:3000/_next/static/chunks/08b5e_@nextui-org_0a72b7._.js:7983:179)
    at http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:693:27
    at runModuleExecutionHooks (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:738:9)
    at instantiateModule (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:691:9)
    at getOrInstantiateModuleFromParent (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:624:12)
    at esmImport (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:142:20)
    at [project]/node_modules/@nextui-org/autocomplete/dist/chunk-KHDVC7JG.mjs [app-client] (ecmascript) (http://localhost:3000/_next/static/chunks/08b5e_@nextui-org_0a72b7._.js:6581:225)
    at http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:693:27
    at runModuleExecutionHooks (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:738:9)
    at instantiateModule (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:691:9)
    at getOrInstantiateModuleFromParent (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:624:12)
    at esmImport (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:142:20)
    at [project]/node_modules/@nextui-org/autocomplete/dist/chunk-KHDVC7JG.mjs [app-client] (ecmascript) <export autocomplete_default as Autocomplete> (http://localhost:3000/_next/static/chunks/08b5e_@nextui-org_0a72b7._.js:6651:184)
    at http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:693:27
    at runModuleExecutionHooks (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:738:9)
    at instantiateModule (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:691:9)
    at getOrInstantiateModuleFromParent (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:624:12)
    at esmImport (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:142:20)
    at [project]/apps/nextjs/src/app/(dashboard)/[workspaceId]/[projectId]/_components/new-post-dialog.tsx [app-client] (ecmascript) (http://localhost:3000/_next/static/chunks/src_e9d541._.js:5782:240)
    at http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:693:27
    at runModuleExecutionHooks (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:738:9)
    at instantiateModule (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:691:9)
    at getOrInstantiateModuleFromParent (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:624:12)
    at esmImport (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:142:20)
    at [project]/apps/nextjs/src/app/(dashboard)/[workspaceId]/[projectId]/_components/sidebar.tsx [app-client] (ecmascript) (http://localhost:3000/_next/static/chunks/src_e9d541._.js:8952:229)
    at http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:693:27
    at runModuleExecutionHooks (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:738:9)
    at instantiateModule (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:691:9)
    at getOrInstantiateModuleFromParent (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:624:12)
    at commonJsRequire (http://localhost:3000/_next/static/chunks/%5Bproject%5D_apps_nextjs_cfb72b._.js:157:20)
    at requireModule (http://localhost:3000/_next/static/chunks/08b5e_next_dist_compiled_1359e6._.js:2290:29)
    at initializeModuleChunk (http://localhost:3000/_next/static/chunks/08b5e_next_dist_compiled_1359e6._.js:2832:25)
    at ProjectLayout (rsc://React/Server/C:%5Cdev%5Cprojects%5Cproject-acme%5Cacme%5Capps%5Cnextjs%5C.next%5Cserver%5Cchunks%5Cssr%5Csrc_5095cc._.js?97:343:278)
    at Router (http://localhost:3000/_next/static/chunks/08b5e_next_dist_client_9bb038._.js:13230:53)
    at AppRouter (http://localhost:3000/_next/static/chunks/08b5e_next_dist_client_9bb038._.js:13269:53)
    at ServerRoot (http://localhost:3000/_next/static/chunks/08b5e_next_dist_client_9bb038._.js:15630:54)

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

Using Autocomplete in latest Release with React 19

### Expected behavior

no forwardRef error

### Screenshots or Videos

_No response_

### Operating System Version

Windows / macOS

### Browser

Chrome

### PR DESCRIPTION
**Pull request by winchesHe**
Closes #4167

## 📝 Description

image

## Summary by CodeRabbit

- **New Features**
	- Added a new "Forms" section to the documentation.
  
- **Updates**
	- Numerous packages under the `@nextui-org` namespace received patch updates, enhancing functionality and performance.
	- Documentation for various components has been updated to reflect recent changes.

- **Bug Fixes**
	- Improved ref handling in several UI components, ensuring better integration with React's ref system.

### CODE REVIEW

File: packages/components/dropdown/src/dropdown-trigger.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export interface DropdownTriggerProps {
  * DropdownTrigger opens the popover's content. It must be an interactive element
  * such as `button` or `a`.
  */
-const DropdownTrigger = forwardRef<"button", DropdownTriggerProps>((props, _) => {
+const DropdownTrigger = wrapForwardRefTypes<"button", DropdownTriggerProps>((props, _) => {
```
Comment (wingkwong) [inline packages/components/dropdown/src/dropdown-trigger.tsx:14]:
```tsx
const DropdownTrigger = wrapForwardRefTypes<"button", DropdownTriggerProps>((props) => {
```

**Review by wingkwong — COMMENTED** [review]
can you share some background context like how to test the original issue? is this for nextjs15 / react19? also please add some test cases if possible.

**Comment by jrgarciadev** [comment]
@winchesHe I think in this case we can just avoid using forward at all instead of mocking the types, just return the component with its props

**Comment by winchesHe** [comment]
> @winchesHe I think in this case we can just avoid using forward at all instead of mocking the types, just return the component with its props

You are right 👍, fixed

**Review by wingkwong — COMMENTED** [review]
@winchesHe please remove those changeset that weren't made by you

**Pull request closed without merging**